.github/workflows/ci.yml

```diff
@@ -94,7 +94,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        version:
+        compiler:
           - clang-3
```

Comment:
clang-3 doesn't work, gcc is used instead.
```
Setting up clang-3.9 (1:3.9.1-19ubuntu1) ...
...
> cmake -B/root/aws-iot-device-sdk-cpp-v2/build/aws-iot-device-sdk-cpp-v2 -H/root/aws-iot-device-sdk-cpp-v2 -DAWS_WARNINGS_ARE_ERRORS=ON -DPERFORM_HEADER_CHECK=ON -DCMAKE_INSTALL_PREFIX=/root/aws-iot-device-sdk-cpp-v2/build/install -DCMAKE_PREFIX_PATH=/root/aws-iot-device-sdk-cpp-v2/build/install -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=ON --no-warn-unused-cli -DPERFORM_HEADER_CHECK=OFF -DS2N_NO_PQ_ASM=ON
...
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
```

Just a thought: maybe it's because we specify v3, but v3.9 is installed?

Reply:
failed had to revert

Reply:
during the installation we go this
```
Setting up libobjc-7-dev:amd64 (7.5.0-3ubuntu1~18.04) ...
Setting up llvm-3.9-dev (1:3.9.1-19ubuntu1) ...
Setting up clang-3.9 (1:3.9.1-19ubuntu1) ...
Processing triggers for libc-bin (2.27-3ubuntu1.6) ...
Processing triggers for mime-support (3.60ubuntu1) ...
WARNING: C compiler clang could not be found for export
WARNING: CXX compiler clang could not be found for export
Finished: InstallCompiler
Running: InstallPackages
Installing packages via apt: build-essential
> sudo apt-get -qq install -y build-essential
```

Reply:
For history: we agreed to handle this in the followup PRs